source/SH3/system/glprogram.cpp

```diff
@@ -122,6 +122,86 @@ void program::Unbind()
     glUseProgram(0); // The 'correct' way to unbind a shader, but apparently undefined?!?!
 }
 
+GLint program::GetUniformLoc(const std::string& name)
+{
+    GLint loc;
+
+    loc = glGetUniformLocation(programID, reinterpret_cast<const GLchar*>(name.c_str()));
+    if(loc == -1)
+        Log(LogLevel::WARN, "program::GetUniformLoc( ): loc == -1! Perhaps the uniform does not exist (check your GLSL source!!!!)");
+
+    return loc;
+}
+
+void program::SetUniformMat2(const std::string& name, const glm::mat2& mat)
+{
+    GLint uniformLoc;
+
+    uniformLoc = GetUniformLoc(name);
+    glUniformMatrix2fv(uniformLoc, 1, GL_FALSE, &mat[0][0]);
```

Comment:
These should use `glm::value_ptr(mat)` instead of `&mat[0][0]`. Same for `glm::vec*`.